docs/ref/forms/modelform-meta.txt

```diff
@@ -0,0 +1,283 @@
+==========================
+ModelForm ``Meta`` options
+==========================
+
+.. module:: django.forms.models
+    :synopsis: ModelForm inner Meta class options
+
+
+.. admonition:: Other ModelForm-related documentation
+
+    This document enumerates and describes the options that can be used to
+    structure a :class:`~django.forms.ModelForm` through its inner ``Meta``
+    class. For additional context, examples and use cases for the
+    :class:`~django.forms.ModelForm` class, please refer to the documentation
+    topic :doc:`Creating forms from models </topics/forms/modelforms>` and the
+    reference entry for :doc:`ModelForm functions</ref/forms/models>`.
+
+.. currentmodule:: django.forms
+
+.. class:: ModelFormOptions
+
+The :class:`~django.forms.ModelForm` helper class can be used to build forms
+that closely reflect specific Django models. The exact structure of the form
+produced can be controlled and customized by defining options in an inner
+``Meta`` class.
+
+.. _ref-modelform-meta-modelformoptions:
+
+``Meta`` and ``ModelFormOptions``
```

Comment:
I'd avoid references to `ModelFormOptions` as mention here would make it public API, which according to this [`FIXME`](https://github.com/django/django/blob/5a91ad3d7115c692d497663a155edee5ebc8989c/django/forms/models.py#L612) it is not.

Reply:
Hmm... I had interpreted that FIXME as applying specifically to the `modelform_factory` function, that is, it _should be_ unnecessary to create a `Meta` class in this function since all the `Meta` attributes that are extracted from a `ModelForm`'s inner `Meta` class (when taking the class definition route to specify a `ModelForm`) have already been provided to this function as arguments.

In support of a narrow interpretation of that FIXME, I would add that `ModelForm.ModelFormOptions` does parallel the design of `Model.Options` (which has its own doc reference page).

I can see your interpretation as well, though, and I will defer to your judgment on this if my reasoning doesn't change your mind!

Reply:
I see what you mean. I'd misunderstood the FIXME. Still the very act of documenting `ModelFormOptions` would change it from private (undocumented and internal) to public.

Reply:
Please note that even [in the docs for Model Meta options](https://docs.djangoproject.com/en/5.1/ref/models/options/) there's no mention of `models.options.Options`, which technically remains private for that reason.

Reply:
Oh, perhaps I'm misunderstanding what you mean by removing references to `ModelFormOptions` which I took to mean removing _any_ mention of it, but now I'm understanding to mean to remove references to the `django.forms.ModelFormOptions` class.

Is keeping it for the attributes (like how the Model Meta options has `Options.abstract`, `Options.app_label` etc.) still OK?